src/nc/core/likec/CxxDocument.cpp

```diff
@@ -80,8 +80,8 @@ inline const TreeNode *getNode(const RangeNode *rangeNode) {
 
 } // anonymous namespace
 
-CxxDocument::CxxDocument(std::shared_ptr<const core::Context> context):
-    context_(std::move(context))
```

Comment:
I don't know if the `std::move` here should go in the constructor of `nc::gui::CxxDocument`.